python/ray/train/v2/_internal/execution/torch_without_ray_train_controller.py

```diff
@@ -0,0 +1,100 @@
+import logging
+from typing import Any, Callable, Dict, Optional
+
+from ray.data import DataIterator
+from ray.train import Checkpoint, Result
+from ray.train.trainer import GenDataset
+from ray.train.v2._internal.execution.train_fn_utils import (
+    TrainFnUtils,
+    get_train_fn_utils,
+    set_train_fn_utils,
+)
+from ray.train.v2._internal.util import date_str
+from ray.train.v2.api.context import (
+    TrainContext as ExternalTrainContext,
+    TrainContextWithoutRayTrainController,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class TorchWithoutRayTrainControllerFnUtils(TrainFnUtils):
+    """TrainFnUtils for jobs launched without ray train controller.
+    This is more for testing purposes, and some functionality is missing.
```

Comment:
It's already mentioned that this is just for testing purposes, but I don't think we plan on having any Torch-specific logic (yet?).  

I think it'd be better if this logic is moved to a common location, e.g. in `DataParallelTrainer.fit()`